issue_comment.go

```diff
@@ -135,18 +135,21 @@ func extractProwJobURLFromCommentBody(logger zerolog.Logger, commentBody string)
 	var prowJobURL string
 
 	r, _ := regexp.Compile(regexToFetchProwURL)
-	sliceOfMatchingString := r.FindStringSubmatch(commentBody)
+	sliceOfMatchingString := r.FindAllStringSubmatch(commentBody, -1)
 	if sliceOfMatchingString == nil {
 		return "", matchNotFoundError
 	}
-	for _, match := range sliceOfMatchingString {
-		if !strings.Contains(match, "images") && !strings.HasSuffix(match, ")") {
-			prowJobURL = match
+	for _, matchesAndGroups := range sliceOfMatchingString {
+		for _, subsStr := range matchesAndGroups {
+			if !strings.Contains(subsStr, "images") && !strings.HasSuffix(subsStr, ")") {
+				prowJobURL = subsStr
+			}
 		}
 	}
 	if prowJobURL == "" {
 		return "", matchNotFoundError
 	}
+
 	logger.Debug().Msgf("Prow Job's URL: %s", prowJobURL)
```

Comment:
nice catch. I just have an idea about simplifying this function a bit
```go
func extractProwJobURLFromCommentBody(commentBody string) (string, error) {
	r, _ := regexp.Compile(regexToFetchProwURL)
	sliceOfMatchingString := r.FindAllStringSubmatch(commentBody, -1)

	for _, matchesAndGroups := range sliceOfMatchingString {
		for _, subsStr := range matchesAndGroups {
			if !strings.Contains(subsStr, "images") && !strings.HasSuffix(subsStr, ")") {
				return subsStr, nil
			}
		}
	}

	return "", fmt.Errorf("regex string %s found no matches for the comment body: %s", regexToFetchProwURL, commentBody)
}
```

and then moving this line [here](https://github.com/redhat-appstudio-qe/ci-helper-app/pull/22/files#diff-078abda156a57dc03e04b099c3a75911a5e460f9ed0fb4e0a4066967c5b4e7bdL100) instead, wdyt?

Reply:
i like the refactoring. curious about the new place, is there any specific reason why you chose that line instead of [this](https://github.com/redhat-appstudio-qe/ci-helper-app/pull/22/files#diff-078abda156a57dc03e04b099c3a75911a5e460f9ed0fb4e0a4066967c5b4e7bdR90) (which is right after this function call returns back successfully)?

Reply:
bah, sorry, yeah it should be the line you mentioned.

Reply:
thanks! updated.